src/cert/client_cert_store.cpp

```diff
@@ -71,3 +104,26 @@ std::string mp::ClientCertStore::PEM_cert_chain() const
         return mp::utils::contents_of(path);
     return {};
 }
+
+bool mp::ClientCertStore::verify_cert(const std::string& pem_cert)
+{
+    // The first client to connect will automatically have its certificate stored
+    if (authenticated_client_certs.empty())
+    {
+        add_cert(pem_cert);
+        return true;
+    }
```

Comment:
This should only happen if the connection is over a restricted unix socket.

Reply:
Ok, I missed the fact the automatic first client to connect was **only** on a restricted unix socket.  I will need to redo this logic.  So for TCP sockets, we will certainly need the `multipassd set <passphrase>` stuff since there would be a chance *no* client could connect.  I know we have this solved on Windows with the install the certs on package install, but in the longer run where all platforms would support TCP.

Reply:
Yeah this was a late change, and you're right that's what we need the "recovery mode" for, even for Windows if things go wrong (the client loses their only authorized credentials).

Reply:
This should be fixed now.